platform/backend/core/src/main/kotlin/io/hamal/core/adapter/Trigger.kt

```diff
@@ -135,6 +145,20 @@ class TriggerAdapter(
         return responseHandler(triggers, funcs, flows, topics, hooks)
     }
 
+    override fun <T : Any> invoke(
+        triggerId: TriggerId,
+        triggerStatus: TriggerStatus,
+        responseHandler: (TriggerStatusSubmitted) -> T
+    ): T {
+        triggerQueryRepository.get(triggerId)  //ensure trigger exists
```

Comment:
Off load this into a private function  - like `ensureFuncExists`
```suggestion
ensureTriggerExists(triggerId)
```